blockchain/client.go

```diff
@@ -26,10 +36,92 @@ func NewClient(url string) (*Client, error) {
 	}
 
 	return &Client{
-		SubstrateAPI: substrateApi,
-		DdcClusters:  pallets.NewDdcClustersApi(substrateApi),
-		DdcCustomers: pallets.NewDdcCustomersApi(substrateApi, meta),
-		DdcNodes:     pallets.NewDdcNodesApi(substrateApi, meta),
-		DdcPayouts:   pallets.NewDdcPayoutsApi(substrateApi, meta),
+		SubstrateAPI:    substrateApi,
+		eventsListeners: make(map[int]EventsListener),
+		DdcClusters:     pallets.NewDdcClustersApi(substrateApi),
+		DdcCustomers:    pallets.NewDdcCustomersApi(substrateApi, meta),
+		DdcNodes:        pallets.NewDdcNodesApi(substrateApi, meta),
+		DdcPayouts:      pallets.NewDdcPayoutsApi(substrateApi, meta),
 	}, nil
 }
+
+func (c *Client) StartEventsListening() (func(), <-chan error, error) {
```

Comment:
Updated. Now we can freely call it twice. It will ignore a duplicate attempt to start events listening (until its stopped) and just returns an existing stopper function and errors channel.